docs/models_and_schedulers.rst

```diff
@@ -252,3 +252,146 @@ the following happens:
   requests.
 
 .. image:: images/sequence_example2.png
+
+.. _section-ensemble-models:
+
+Ensemble Models
+---------------
+
+An ensemble model represents a *pipepline* of one or more models and the
+connection of input and output tensors between those models. Ensemble models are
+intended to be used to encapsulate a procedure that involves
+multiple models, such as "data preprocessing -> inference -> data postprocessing".
+Using ensemble models for this purpose can avoid the overhead of transferring
+intermediate tensors and minimize the number of requests that must be sent to
+the inference server.
+
+The :ref:`ensemble scheduler <section-ensemble-scheduler>` must be used for
+ensemble models, regardless of the scheduler used by the models within the
+ensemble. With respect to the ensemble scheduler, an *ensemble* model is not
+an actual model. Instead, it specifies the dataflow between models within the
+ensemble as :cpp:var:`Step
+<nvidia::inferenceserver::ModelEnsembling::Step>`. The
+scheduler collects the output tensors in each step, provides them as input
+tensors for other steps according to the specification. In spite of that, the
+ensemble model is still viewed as a single model from an external view.
+
+Note that the ensemble models will inherit the characteristics of the models
+involved, so the meta-data in the request header must comply with the models
+within the ensemble. For instance, if one of the models is stateful model, then
+the inference request for the ensemble model should contain the information
+mentioned in the previous :ref:`section <section-stateful-models>`, which will
+be provided to the stateful model by the scheduler. This is a limitation of
```

Comment:
I think you missed my earlier comments on this paragraph. Just mentioning "flag" at the end is confusing.

Reply:
I can't find earlier comments either... strange. I think you should at least drop the last sentence since mentioning flags like that is probably just more confusing.